// vue-baidu-map/components/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.BmlCurveLine = exports.BmlHeatmap = exports.BmlLushu = exports.BmlMarkerClusterer = exports.BmAutoComplete = exports.BmBoundary = exports.BmTraffic = exports.BmTile = exports.BmBus = exports.BmDriving = exports.BmWalking = exports.BmTransit = exports.BmLocalSearch = exports.BmContextMenuItem = exports.BmContextMenu = exports.BmOverlay = exports.BmInfoWindow = exports.BmLabel = exports.BmGround = exports.BmCircle = exports.BmPolygon = exports.BmPolyline = exports.BmPointCollection = exports.BmMarker = exports.BmControl = exports.BmPanorama = exports.BmCityList = exports.BmCopyright = exports.BmGeolocation = exports.BmOverviewMap = exports.BmMapType = exports.BmNavigation = exports.BmScale = exports.BmView = exports.BaiduMap = undefined;

var _Map = require('./map/Map.vue');

var _Map2 = _interopRequireDefault(_Map);

var _MapView = require('./map/MapView.vue');

var _MapView2 = _interopRequireDefault(_MapView);

var _Scale = require('./controls/Scale.vue');

var _Scale2 = _interopRequireDefault(_Scale);

var _Navigation = require('./controls/Navigation.vue');

var _Navigation2 = _interopRequireDefault(_Navigation);

var _MapType = require('./controls/MapType.vue');

var _MapType2 = _interopRequireDefault(_MapType);

var _OverviewMap = require('./controls/OverviewMap.vue');

var _OverviewMap2 = _interopRequireDefault(_OverviewMap);

var _Geolocation = require('./controls/Geolocation.vue');

var _Geolocation2 = _interopRequireDefault(_Geolocation);

var _Copyright = require('./controls/Copyright.vue');

var _Copyright2 = _interopRequireDefault(_Copyright);

var _CityList = require('./controls/CityList.vue');

var _CityList2 = _interopRequireDefault(_CityList);

var _Panorama = require('./controls/Panorama.vue');

var _Panorama2 = _interopRequireDefault(_Panorama);

var _Control = require('./controls/Control.vue');

var _Control2 = _interopRequireDefault(_Control);

var _Marker = require('./overlays/Marker.vue');

var _Marker2 = _interopRequireDefault(_Marker);

var _PointCollection = require('./overlays/PointCollection.vue');

var _PointCollection2 = _interopRequireDefault(_PointCollection);

var _Polyline = require('./overlays/Polyline.vue');

var _Polyline2 = _interopRequireDefault(_Polyline);

var _Polygon = require('./overlays/Polygon.vue');

var _Polygon2 = _interopRequireDefault(_Polygon);

var _Circle = require('./overlays/Circle.vue');

var _Circle2 = _interopRequireDefault(_Circle);

var _Ground = require('./overlays/Ground.vue');

var _Ground2 = _interopRequireDefault(_Ground);

var _Label = require('./overlays/Label.vue');

var _Label2 = _interopRequireDefault(_Label);

var _InfoWindow = require('./overlays/InfoWindow.vue');

var _InfoWindow2 = _interopRequireDefault(_InfoWindow);

var _Overlay = require('./overlays/Overlay.vue');

var _Overlay2 = _interopRequireDefault(_Overlay);

var _Menu = require('./context-menu/Menu.vue');

var _Menu2 = _interopRequireDefault(_Menu);

var _Item = require('./context-menu/Item.vue');

var _Item2 = _interopRequireDefault(_Item);

var _LocalSearch = require('./search/LocalSearch.vue');

var _LocalSearch2 = _interopRequireDefault(_LocalSearch);

var _Transit = require('./search/Transit.vue');

var _Transit2 = _interopRequireDefault(_Transit);

var _Walking = require('./search/Walking.vue');

var _Walking2 = _interopRequireDefault(_Walking);

var _Driving = require('./search/Driving.vue');

var _Driving2 = _interopRequireDefault(_Driving);

var _Bus = require('./search/Bus.vue');

var _Bus2 = _interopRequireDefault(_Bus);

var _Tile = require('./layers/Tile.vue');

var _Tile2 = _interopRequireDefault(_Tile);

var _Traffic = require('./layers/Traffic.vue');

var _Traffic2 = _interopRequireDefault(_Traffic);

var _Boundary = require('./others/Boundary.vue');

var _Boundary2 = _interopRequireDefault(_Boundary);

var _AutoComplete = require('./others/AutoComplete.vue');

var _AutoComplete2 = _interopRequireDefault(_AutoComplete);

var _MarkerClusterer = require('./extra/MarkerClusterer.vue');

var _MarkerClusterer2 = _interopRequireDefault(_MarkerClusterer);

var _Lushu = require('./extra/Lushu.vue');

var _Lushu2 = _interopRequireDefault(_Lushu);

var _Heatmap = require('./extra/Heatmap.vue');

var _Heatmap2 = _interopRequireDefault(_Heatmap);

var _CurveLine = require('./extra/CurveLine.vue');

var _CurveLine2 = _interopRequireDefault(_CurveLine);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
  install: function install(Vue, options) {
    var ak = options.ak;

    Vue.prototype._BMap = function () {
      return { ak: ak };
    };

    Vue.component('baidu-map', _Map2.default);
    Vue.component('bm-view', _MapView2.default);

    Vue.component('bm-scale', _Scale2.default);
    Vue.component('bm-navigation', _Navigation2.default);
    Vue.component('bm-map-type', _MapType2.default);
    Vue.component('bm-overview-map', _OverviewMap2.default);
    Vue.component('bm-geolocation', _Geolocation2.default);
    Vue.component('bm-copyright', _Copyright2.default);
    Vue.component('bm-city-list', _CityList2.default);
    Vue.component('bm-panorama', _Panorama2.default);
    Vue.component('bm-control', _Control2.default);

    Vue.component('bm-marker', _Marker2.default);
    Vue.component('bm-point-collection', _PointCollection2.default);
    Vue.component('bm-polyline', _Polyline2.default);
    Vue.component('bm-polygon', _Polygon2.default);
    Vue.component('bm-circle', _Circle2.default);
    Vue.component('bm-ground', _Ground2.default);
    Vue.component('bm-label', _Label2.default);
    Vue.component('bm-info-window', _InfoWindow2.default);
    Vue.component('bm-overlay', _Overlay2.default);

    Vue.component('bm-context-menu', _Menu2.default);
    Vue.component('bm-context-menu-item', _Item2.default);

    Vue.component('bm-local-search', _LocalSearch2.default);
    Vue.component('bm-transit', _Transit2.default);
    Vue.component('bm-walking', _Walking2.default);
    Vue.component('bm-driving', _Driving2.default);
    Vue.component('bm-bus', _Bus2.default);

    Vue.component('bm-tile', _Tile2.default);
    Vue.component('bm-traffic', _Traffic2.default);

    Vue.component('bm-auto-complete', _AutoComplete2.default);
    Vue.component('bm-boundary', _Boundary2.default);
  }
};
exports.BaiduMap = _Map2.default;
exports.BmView = _MapView2.default;
exports.BmScale = _Scale2.default;
exports.BmNavigation = _Navigation2.default;
exports.BmMapType = _MapType2.default;
exports.BmOverviewMap = _OverviewMap2.default;
exports.BmGeolocation = _Geolocation2.default;
exports.BmCopyright = _Copyright2.default;
exports.BmCityList = _CityList2.default;
exports.BmPanorama = _Panorama2.default;
exports.BmControl = _Control2.default;
exports.BmMarker = _Marker2.default;
exports.BmPointCollection = _PointCollection2.default;
exports.BmPolyline = _Polyline2.default;
exports.BmPolygon = _Polygon2.default;
exports.BmCircle = _Circle2.default;
exports.BmGround = _Ground2.default;
exports.BmLabel = _Label2.default;
exports.BmInfoWindow = _InfoWindow2.default;
exports.BmOverlay = _Overlay2.default;
exports.BmContextMenu = _Menu2.default;
exports.BmContextMenuItem = _Item2.default;
exports.BmLocalSearch = _LocalSearch2.default;
exports.BmTransit = _Transit2.default;
exports.BmWalking = _Walking2.default;
exports.BmDriving = _Driving2.default;
exports.BmBus = _Bus2.default;
exports.BmTile = _Tile2.default;
exports.BmTraffic = _Traffic2.default;
exports.BmBoundary = _Boundary2.default;
exports.BmAutoComplete = _AutoComplete2.default;
exports.BmlMarkerClusterer = _MarkerClusterer2.default;
exports.BmlLushu = _Lushu2.default;
exports.BmlHeatmap = _Heatmap2.default;
exports.BmlCurveLine = _CurveLine2.default;
